pkg/prometheus/statefulset_test.go

```diff
@@ -780,12 +780,24 @@ func TestThanosNoObjectStorage(t *testing.T) {
 	}
 
 	if sset.Spec.Template.Spec.Containers[2].Name != "thanos-sidecar" {
-		t.Fatalf("expected 3rd containers to be thanos-sidecar, got %s", sset.Spec.Template.Spec.Containers[2].Name)
+		t.Fatalf("expected 3rd container to be thanos-sidecar, got %s", sset.Spec.Template.Spec.Containers[2].Name)
```

Comment:
Should we change this to check if any of the containers is `thanos-sidecar` and not rely on order?